synapse/app/homeserver.py

```diff
@@ -361,9 +367,37 @@ def setup(config_options):
     logger.info("Database prepared in %s.", config.database_config['name'])
 
     hs.setup()
-    hs.start_listening()
+    # If configured, start up the ACME listener. We will need to check if we
+    # have a certificate, first.
+    if hs.config.acme_enabled:
+        acme = hs.get_acme_handler()
+        acme.start_listening()
+    else:
+        hs.start_listening()
 
     def start():
+        if hs.config.acme_enabled:
+            is_valid_cert = acme.is_disk_cert_valid()
+            if not is_valid_cert:
```

Comment:
Yeah, that was one of the next little thinkies that I have to tackle. I was going to do it in the "enable cert reloads without restarting", since it  means I have to _stop_ things listening and then start them again, which I think is making this a bit too big of a PR.

Reply:
as it stands, it will renew if there's less than 10 days left, so it shouldn't boot with a cert which will immediately expire. Of course, that's not a permanent solution, but it gives people who want to immediately use the feature in the meantime some headroom.